ui/src/app/cohort-search/redux/actions/service.ts

```diff
@@ -119,6 +123,35 @@ export class CohortSearchActions {
     this.addId(newId);
     return newId;
   }
+  updateTemporal(flag: boolean, groupId: string, role: keyof SearchRequest) {
+    this._updatedTemporal(flag, groupId);
+    const group = getGroup(groupId)(this.state);
+    if (this.hasActiveItems(group)) {
+      this.requestGroupCount(role, groupId);
+      this.requestTotalCount(groupId);
+    } else {
+      this.clearGroupCount(groupId);
+      this.clearTotalCount(groupId);
+    }
+  }
+
+  updateWhichMention(mention: any, groupId: string, role: keyof SearchRequest) {
+    this._updateWhichMention(mention, groupId);
+    this.requestGroupCount(role, groupId);
+    this.requestTotalCount(groupId);
+  }
+
+  updateTemporalTime(time: any, groupId: string, role: keyof SearchRequest) {
+    this._updateTemporalTime(time, groupId);
+    this.requestGroupCount(role, groupId);
+    this.requestTotalCount(groupId);
+  }
+
+  updateTemporalTimeValue(timeValue: any, groupId: string, role: keyof SearchRequest) {
+    this._updateTemporalTimeValue(timeValue, groupId);
+    this.requestGroupCount(role, groupId);
+    this.requestTotalCount(groupId);
+  }
```

Comment:
We need to check both temporalGroups for active items in each of the above functions before making the calls for count. Probably need to make a separate function since we'll use it in several actions

Reply:
Done